README.md

```diff
@@ -208,6 +213,183 @@ try {
+
+<a name="messaging"/>
+
+## Trace messages
+
+You can use the SDK to trace messages sent or received via messaging system. When tracing messages, we distinct between:
+* sending a message
+* receiving a message
+* processing a received message
+
+To trace an outgoing message, the code looks straight forward compared to other tracers:
+ 
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+OutgoingMessageTracer outgoingMessageTracer = oneAgentSDK.traceOutgoingMessage(messagingSystemInfo);
+outgoingMessageTracer.start();
+try {
+	// transport the dynatrace tag along with the message: 	
+	messageToSend.setHeaderField(
+		OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME, outgoingMessageTracer.getDynatraceStringTag());
+	theQueue.send(messageToSend);
+	
+	// optional:  add messageid provided from messaging system
+	outgoingMessageTracer.setVendorMessageId(toSend.getMessageId());
+	// optional:  add correlationId
+	outgoingMessageTracer.setCorrelationId(toSend.correlationId);
+} catch (Exception e) {
+	outgoingMessageTracer.error(e.getMessage());
+	Logger.logError(e);
+} finally {
+	outgoingMessageTracer.end();
+}
+```
+
+On the incoming side, we need to differ between the blocking receiving part and processing the received message. Therefore two different tracers are being used: ReceivingMessageTracer and ProcessingMessageTracer.
+
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+// following code is probably inside a busy loop:
+while(true) {
+	ReceivingMessageTracer receivingMessageTracer = oneAgentSDK.traceReceivingMessage(messagingSystemInfo);
+	receivingMessageTracer.start();
+	try {
+		// blocking call - until message is being available:
+		Message queryMessage = theQueue.receive("client queries");
+		ProcessingMessageTracer processingMessageTracer = oneAgentSDK
+			.traceProcessingMessage(messagingSystemInfo);
+		processingMessageTracer.setDynatraceStringTag(
+			queryMessage.getHeaderField(OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME));
+		processingMessageTracer.setVendorMessageId(queryMessage.msgId);
+		processingMessageTracer.setCorrelationId(queryMessage.correlationId);
+		processingMessageTracer.start();
+		try {
+			// do the work ... 
+		} catch (Exception e) {
+			processingMessageTracer.error(e.getMessage());
+			Logger.logError(e);
+		} finally {
+			processingMessageTracer.end();
+		}
+	} catch (Exception e) {
+		receivingMessageTracer.error(e.getMessage());
+		Logger.logError(e);
+	} finally {
+		receivingMessageTracer.end();
+	}
+}
+```
+
+In case of no blocking call happens (e. g. message processing via eventhandler), there is no need to use ReceivingMessageTracer - just trace processing of the message by using the ProcessingMessageTracer:
```

Comment:
I have the feeling that wording could be improved. Maybe something like 'In case of non blocking receive (e.g. via event handler)'.
ProcessingMessageTracer => `fixed font`